grommet/grommet
# #18027 Fix target is not writable on the native event

#### What does this PR do?

Fixing select event behavior in Preact by changing simple assignment to `Object.defineProperty`.

#### Where should the reviewer start?

1. Understand the bug's context from [the issue](https://github.com/grommet/grommet/issues/4340).
2. Check [the sandbox preview](https://codesandbox.io/s/grommet-preact-select-fix-4zmlx).
3. Go through the code.

#### What testing has been done on this PR?

- Open a select.
- Click on a value.
- Be sure that the value is set.

#### How should this be manually tested?

Go to the sandbox https://codesandbox.io/s/grommet-preact-select-fix-4zmlx and check that the select value is changing without errors.
This sandbox patching the grommet package with the change from this PR.

#### Any background context you want to provide?

Inspired by a similar solution for the `material-ui` package: https://github.com/mui-org/material-ui/pull/18027/files.

#### What are the relevant issues?

https://github.com/grommet/grommet/issues/4340

#### Screenshots (if appropriate)

---

#### Do the grommet docs need to be updated?

No.

#### Should this PR be mentioned in the release notes?

No.

#### Is this change backwards compatible or is it a breaking change?

It's backward compatible.
